pkg/client/registration.go

```diff
@@ -117,29 +117,31 @@ func registerAsMemberCandidate(
 	// We do the same in case the registration of eligible operator failed for
 	// some reason. As soon as the operator is eligible, we will proceed with
 	// the registration.
-	registerAsMemberCandidateWhenEligible(parentCtx, ethereumChain, application)
+	registerAsMemberCandidateWhenEligible(parentCtx, blockCounter, application)
 }
 
 // registerAsMemberCandidateWhenEligible for each new block checks the operator's
 // eligibility to be registered as a keep member candidate for the application.
 // As soon as the operator becomes eligible, function triggers the registration.
+//
+// TODO Move this to application.RegisterAsMemberCandidateWhenEligible() <-chan error?
```

Comment:
I'd be fine either way TBH. This file contains the logic about monitoring and updating the status, the registration itself, so `registerAsMemberCandidateWhenEligible` fits well here as well in my opinion.

Reply:
One of the reasons I was thinking about this is that `registerAsMemberCandidateWhenEligible` is what requires the block counter, which as we've discussed I'd like to think about making chain-internal concept, to be threaded through every function in this file. It also doesn't seem like a crazy thing to have the application handle be conceptually responsible for. But I'm not ready to commit to the approach :grimacing:

Reply:
I see, good note about the block counter, I did not take it into account.